### PR TITLE
feature: copy cURL

### PR DESCRIPTION
**Pull request by SyifaAinnur**
## Description

Added copy curl function to detail inspector according to response

example of curl results

GET :
curl -i \
	-d "{userId: 1, id: 1, title: sunt aut facere repellat provident occaecati excepturi optio reprehenderit, body: quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto}" \
	"https://jsonplaceholder.typicode.com/posts/1"
	
POST :
curl -i \
	-X POST \
	-H "content-type: text/plain; charset=utf-8" \
	-d "{id: 101}" \
	"https://jsonplaceholder.typicode.com/posts"
	
PUT :
curl -i \
	-X PUT \
	-H "content-type: application/x-www-form-urlencoded; charset=utf-8" \
	-d "{title: PUT foo, body: PUT bar, userId: 101010, id: 1}" \
	"https://jsonplaceholder.typicode.com/posts/1"
	
	
image

## Type of Change

- [x ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Comment by bwcr** [comment]
This is a great way to handle things when testers discover problems and pass them on to developers for more explanation.

**Comment by syedmurtaza108** [comment]
@SyifaAinnur Thank you adding this PR. Can you please look into the failed action (https://github.com/syedmurtaza108/chucker-flutter/pull/87/checks). Flutter analyze command is failing. Mostly analayzer issues can be fixed using `dart fix --apply` command.

**Comment by SyifaAinnur** [comment]
@syedmurtaza108  how  to fix the failed action in flutter test ?

**Comment by SyifaAinnur** [comment]
@syedmurtaza108 how to fix the failed action in test ?, can you help me how to resolve it ?

**Comment by syedmurtaza108** [comment]
@SyifaAinnur I believe some actions are failing because of Flutter's version used in this repo. I will look into it. Don't worry.

File: lib/src/interceptors/chopper_interceptor.dart [hunk 1]
```diff
@@ -29,10 +29,12 @@ class ChuckerChopperInterceptor implements Interceptor {
         statusCode: response.statusCode,
         connectionTimeout: 0,
         contentType: _requestType(response),
-        headers: response.base.headers.toString(),
-        queryParameters:
-            response.base.request?.url.queryParameters.toString() ??
-                emptyString,
+        // headers: response.base.headers.toString(),
```
Comment (syedmurtaza108) [inline lib/src/interceptors/chopper_interceptor.dart:32]: Can you please make sure that no commented out code is left in the PR.

File: lib/src/view/api_detail_page.dart [hunk 2]
```diff
@@ -126,6 +138,42 @@ class _ApiDetailsPageState extends State<ApiDetailsPage> {
   }
 }
 
+String _cURLRepresentation(ApiResponse api) {
```
Comment (syedmurtaza108) [inline lib/src/view/api_detail_page.dart:141]: Can you please write some test cases for cURL feature?

**Review by syedmurtaza108 — CHANGES_REQUESTED** [review]

**Comment by syedmurtaza108** [comment]
@SyifaAinnur Please add widget and unit tests for the feature you added. See codecov comment https://github.com/syedmurtaza108/chucker-flutter/pull/87#issuecomment-2342530464.

**Comment by syedmurtaza108** [comment]
@SyifaAinnur Flutter analyze command failed.

**Review by syedmurtaza108 — APPROVED** [review]

**Comment by syedmurtaza108** [comment]
@SyifaAinnur Thanks for adding this feature. As a token of appreciation, I have added you in the README file of this repo.

**Pull request merged**